lib/internal/util/debuglog.js

```diff
@@ -17,20 +16,19 @@ const { inspect, format, formatWithOptions } = require('internal/util/inspect');
 // debuglog() before initializeDebugEnv() is called will throw.
 let debugImpls;
 
-let debugEnvRegex = /^$/;
-let testEnabled;
+let testEnabled = () => false;
 
 // `debugEnv` is initial value of process.env.NODE_DEBUG
 function initializeDebugEnv(debugEnv) {
   debugImpls = ObjectCreate(null);
   if (debugEnv) {
+    // This is run before any user code, it's OK not to use primordials.
     debugEnv = debugEnv.replace(/[|\\{}()[\]^$+?.]/g, '\\$&')
-      .replace(/\*/g, '.*')
-      .replace(/,/g, '$|^')
-      .toUpperCase();
-    debugEnvRegex = new RegExp(`^${debugEnv}$`, 'i');
+      .replaceAll('*', '.*')
+      .replaceAll(',', '$|^');
+    const debugEnvRegex = new RegExp(`^${debugEnv}$`, 'i');
+    testEnabled = (str) => RegExpPrototypeExec(debugEnvRegex, str) !== null;
```

Comment:
So, `regexp.exec(str) !== null` is faster than `regexp.test(str)` ?

Reply:
I'm not sure if it's faster, but `RegExpPrototypeTest` looks up `.exec` property of the regex, which is user-mutable (see https://github.com/nodejs/node/pull/38635/files#diff-1fd7756e2222fdc2c8dbd60f718237a78233a8697d418ec9044e167b673bcf96R544-R551).